clerk/middleware_v2.go

```diff
@@ -130,13 +133,26 @@ func WithSessionV2(client Client, verifyTokenOptions ...VerifyTokenOption) func(
 			}
 
 			claims, err := client.VerifyToken(cookieToken.Value, verifyTokenOptions...)
-			if err == nil && claims.IssuedAt != nil && clientUatTs <= int64(*claims.IssuedAt) {
-				ctx := context.WithValue(r.Context(), ActiveSessionClaims, claims)
-				next.ServeHTTP(w, r.WithContext(ctx))
+
+			if err == nil {
+				if claims.IssuedAt != nil && clientUatTs <= int64(*claims.IssuedAt) {
+					ctx := context.WithValue(r.Context(), ActiveSessionClaims, claims)
+					next.ServeHTTP(w, r.WithContext(ctx))
+					return
+				}
+
+				renderInterstitial(client, w)
```

Comment:
❓ Is there is no error during token verification, why are we still rendering the interstitial?

Reply:
Because of this condition in Interstitial flow diagram:
<img width="715" alt="Screenshot 2023-03-01 at 16 48 37" src="https://user-images.githubusercontent.com/3936408/222174556-8aa53d79-8bf1-421e-9280-b30141a73bdb.png">

Reply:
@chanioxaris does this makes any sense ?

Reply:
Yes, thanks for the explanation